tests/packages.py

```diff
@@ -0,0 +1,112 @@
+        that this doesn't fall over.
+        """
+        lsst.base.getEnvironmentPackages()
+
+    def testRuntime(self):
+        """Test getting versions from runtime libraries
+
+        Unfortunately, none of the products that we get runtime versions from are
+        dependencies of this package, and so all we can do is test that this doesn't
+        fall over.
+        """
+        lsst.base.getRuntimeVersions()
+
+    def testPackages(self):
+        """Test the Packages class"""
+        packages = lsst.base.Packages.fromSystem()
+
+        # Test pickling
+        # Can't use lsst.utils.tests.getTempFilePath because we're its dependency
+        temp = tempfile.NamedTemporaryFile(prefix="packages.", suffix=".pickle", delete=False)
+        tempName = temp.name
+        temp.close()  # We don't use the fd, just want a filename
+        try:
+            packages.write(tempName)
+            new = lsst.base.Packages.read(tempName)
+        finally:
+            os.unlink(tempName)
+
+        # 'packages' and 'new' should have identical content
+        self.assertDictEqual(packages.difference(new), {})
+        self.assertDictEqual(packages.missing(new), {})
+        self.assertDictEqual(packages.extra(new), {})
+        self.assertEqual(len(packages), len(new))
+
+        # Check inverted comparisons
+        self.assertDictEqual(new.difference(packages), {})
+        self.assertDictEqual(new.missing(packages), {})
+        self.assertDictEqual(new.extra(packages), {})
+
+        # Now load an obscure python package and the list of packages should change
+        import xmlrpclib  # Shouldn't be used by anything we've previously imported
+        new = lsst.base.Packages.fromSystem()
+        self.assertDictEqual(packages.difference(new), {})  # No inconsistencies
+        self.assertDictEqual(packages.extra(new), {})  # Nothing in 'packages' that's not in 'new'
+        missing = packages.missing(new)
+        self.assertNotEqual(len(missing), 0)  # 'packages' should be missing some stuff in 'new'
+        self.assertIn("xmlrpclib", missing)
+
+        # Inverted comparisons
+        self.assertDictEqual(new.difference(packages), {})
+        self.assertDictEqual(new.missing(packages), {})  # Nothing in 'new' that's not in 'packages'
+        extra = new.extra(packages)
+        self.assertNotEqual(len(extra), 0)  # 'new' has extra stuff compared to 'packages'
+        self.assertIn("xmlrpclib", extra)
+
+        packages.update(new)  # Should now be identical
+        self.assertDictEqual(packages.difference(new), {})
+        self.assertDictEqual(packages.missing(new), {})
+        self.assertDictEqual(packages.extra(new), {})
+        self.assertEqual(len(packages), len(new))
+
+
+def suite():
+    suites = []
+    suites += unittest.makeSuite(PackagesTestCase)
+    return unittest.TestSuite(suites)
+
+if __name__ == "__main__":
+    sys.exit(unittest.TextTestRunner().run(suite()).wasSuccessful())
```

Comment:
This fails on my Mac:
```
$ python tests/packages.py 
....
----------------------------------------------------------------------
Ran 4 tests in 0.452s

OK
$ echo $?
1
```
So something strange is going on with that exiting code. Please just remove the `suite()` and use `unittest.main()`.